### PR TITLE
[#115] textual representation of log messages - for readability in debugging

### PR DESCRIPTION
**Pull request by CodiePP**
add the flag `--trace-chain-db` to turn on textual representation of logged messages.
(-> in shelley-testnet.sh, variable SPECIAL)

### CODE REVIEW

**Comment by CodiePP** [comment]
example output:
```
[iohk.cardano.node.acceptor:Debug:4] [2019-08-21 13:29:54.74 UTC] **************************************
[iohk.cardano.node.acceptor:Debug:4] [2019-08-21 13:29:54.74 UTC] I am TraceAcceptor
[iohk.cardano.node.acceptor:Debug:4] [2019-08-21 13:29:54.74 UTC] **************************************
[iohk.cardano.node.CoreId 2:Debug:19] [2019-08-21 13:30:12.92 UTC] System started at SystemStart {getSystemStart = 1858-11-17 00:00:00 UTC}
[iohk.cardano.node.CoreId 2:Debug:19] [2019-08-21 13:30:12.92 UTC] **************************************
[iohk.cardano.node.CoreId 2:Debug:19] [2019-08-21 13:30:12.92 UTC] I am Node = NodeAddress {naHostAddress = ::1, naPort = 3002}
[iohk.cardano.node.CoreId 2:Debug:19] [2019-08-21 13:30:12.92 UTC] My producers are [RemoteAddress {raAddress = "::1", raPort = 3000, raValency = 1},RemoteAddress {raAddress = "127.0.0.1", raPort = 3001, raValency = 1}]
[iohk.cardano.node.CoreId 2:Debug:19] [2019-08-21 13:30:12.92 UTC] **************************************
[iohk.cardano.node.CoreId 2:Debug:19] [2019-08-21 13:30:12.92 UTC] [genesis] TraceImmDBEvent
[iohk.cardano.node.CoreId 2:Debug:19] [2019-08-21 13:30:12.92 UTC] [genesis] TraceImmDBEvent
[iohk.cardano.node.CoreId 2:Debug:19] [2019-08-21 13:30:12.92 UTC] [genesis] Opened imm db with immutable tip at Origin and epoch EpochNo {unEpochNo = 0}
[iohk.cardano.node.CoreId 2:Debug:19] [2019-08-21 13:30:12.92 UTC] [genesis] Opened vol db
[iohk.cardano.node.CoreId 2:Debug:19] [2019-08-21 13:30:12.92 UTC] [genesis] Replaying ledger from snapshot DiskSnapshot 2 at genesis
[iohk.cardano.node.CoreId 2:Debug:19] [2019-08-21 13:30:12.92 UTC] [genesis] Opened lgr db
[iohk.cardano.node.CoreId 2:Debug:19] [2019-08-21 13:30:12.92 UTC] [genesis] InitChainSelValidation
[iohk.cardano.node.CoreId 2:Debug:19] [2019-08-21 13:30:12.92 UTC] [genesis] Opened db with immutable tip at Origin and tip (Point 2536552761, 382dbfce73d21b79)
[iohk.cardano.node.CoreId 2:Debug:31] [2019-08-21 13:30:12.92 UTC] [genesis] Took ledger snapshot DiskSnapshot 3 at genesis
[iohk.cardano.node.CoreId 2:Debug:31] [2019-08-21 13:30:12.92 UTC] [genesis] Deleted old snapshot DiskSnapshot 1
[iohk.cardano.node.CoreId 2:Debug:35] [2019-08-21 13:30:18.00 UTC] [382dbfc] Chain added block (Point 2536555509, 179c6d869fd9a438)
[iohk.cardano.node.CoreId 2:Debug:35] [2019-08-21 13:30:18.00 UTC] [382dbfc] Block fits onto the current chain: (Point 2536555509, 179c6d869fd9a438)
[iohk.cardano.node.CoreId 2:Debug:35] [2019-08-21 13:30:18.00 UTC] [382dbfc] Valid candidate (Point 2536555509, 179c6d869fd9a438)
[iohk.cardano.node.CoreId 2:Debug:35] [2019-08-21 13:30:18.00 UTC] [382dbfc] Chain changed, new tip: (Point 2536555509, 179c6d869fd9a438)
[iohk.cardano.node.CoreId 2:Debug:35] [2019-08-21 13:30:26.00 UTC] [179c6d8] Chain added block (Point 2536555513, 21264408360c9b3e)
[iohk.cardano.node.CoreId 2:Debug:35] [2019-08-21 13:30:26.00 UTC] [179c6d8] Block fits onto the current chain: (Point 2536555513, 21264408360c9b3e)
[iohk.cardano.node.CoreId 2:Debug:35] [2019-08-21 13:30:26.00 UTC] [179c6d8] Valid candidate (Point 2536555513, 21264408360c9b3e)
[iohk.cardano.node.CoreId 2:Debug:35] [2019-08-21 13:30:26.00 UTC] [179c6d8] Chain changed, new tip: (Point 2536555513, 21264408360c9b3e)
[iohk.cardano.node.CoreId 2:Debug:31] [2019-08-21 13:30:32.92 UTC] [2126440] Took ledger snapshot DiskSnapshot 4 at genesis
[iohk.cardano.node.CoreId 2:Debug:31] [2019-08-21 13:30:32.92 UTC] [2126440] Deleted old snapshot DiskSnapshot 2
[iohk.cardano.node.CoreId 2:Debug:35] [2019-08-21 13:30:34.00 UTC] [2126440] Chain added block (Point 2536555517, 56bb14559d27ed1f)
[iohk.cardano.node.CoreId 2:Debug:35] [2019-08-21 13:30:34.00 UTC] [2126440] Block fits onto the current chain: (Point 2536555517, 56bb14559d27ed1f)
[iohk.cardano.node.CoreId 2:Debug:35] [2019-08-21 13:30:34.00 UTC] [2126440] Valid candidate (Point 2536555517, 56bb14559d27ed1f)
[iohk.cardano.node.CoreId 2:Debug:35] [2019-08-21 13:30:34.00 UTC] [2126440] Chain changed, new tip: (Point 2536555517, 56bb14559d27ed1f)
[iohk.cardano.node.CoreId 2:Debug:35] [2019-08-21 13:30:42.00 UTC] [56bb145] Chain added block (Point 2536555521, 01d5535358ffc50c)
[iohk.cardano.node.CoreId 2:Debug:35] [2019-08-21 13:30:42.00 UTC] [56bb145] Block fits onto the current chain: (Point 2536555521, 01d5535358ffc50c)
[iohk.cardano.node.CoreId 2:Debug:35] [2019-08-21 13:30:42.00 UTC] [56bb145] Valid candidate (Point 2536555521, 01d5535358ffc50c)
[iohk.cardano.node.CoreId 2:Debug:35] [2019-08-21 13:30:42.00 UTC] [56bb145] Chain changed, new tip: (Point 2536555521, 01d5535358ffc50c)
```

**Comment by CodiePP** [comment]
example output with `--tracing-verbosity-maximal`:
```
[iohk.cardano.node.CoreId 2:Debug:19] [2019-08-21 14:06:08.97 UTC] System started at SystemStart {getSystemStart = 1858-11-17 00:00:00 UTC}
[iohk.cardano.node.CoreId 2:Debug:19] [2019-08-21 14:06:08.97 UTC] **************************************
[iohk.cardano.node.CoreId 2:Debug:19] [2019-08-21 14:06:08.97 UTC] I am Node = NodeAddress {naHostAddress = ::1, naPort = 3002}
[iohk.cardano.node.CoreId 2:Debug:19] [2019-08-21 14:06:08.97 UTC] My producers are [RemoteAddress {raAddress = "::1", raPort = 3000, raValency = 1},RemoteAddress {raAddress = "127.0.0.1", raPort = 3001, raValency = 1}]
[iohk.cardano.node.CoreId 2:Debug:19] [2019-08-21 14:06:08.97 UTC] **************************************
[iohk.cardano.node.CoreId 2:Debug:19] [2019-08-21 14:06:08.97 UTC] [genesis] TraceImmDBEvent
[iohk.cardano.node.CoreId 2:Debug:19] [2019-08-21 14:06:08.97 UTC] [genesis] TraceImmDBEvent
[iohk.cardano.node.CoreId 2:Debug:19] [2019-08-21 14:06:08.97 UTC] [genesis] Opened imm db with immutable tip at Origin and epoch EpochNo {unEpochNo = 0}
[iohk.cardano.node.CoreId 2:Debug:19] [2019-08-21 14:06:08.97 UTC] [genesis] Opened vol db
[iohk.cardano.node.CoreId 2:Debug:19] [2019-08-21 14:06:08.97 UTC] [genesis] Replaying ledger from snapshot DiskSnapshot 5 at genesis
[iohk.cardano.node.CoreId 2:Debug:19] [2019-08-21 14:06:08.97 UTC] [genesis] Opened lgr db
[iohk.cardano.node.CoreId 2:Debug:19] [2019-08-21 14:06:08.97 UTC] [genesis] InitChainSelValidation
[iohk.cardano.node.CoreId 2:Debug:19] [2019-08-21 14:06:08.97 UTC] [genesis] Opened db with immutable tip at Origin and tip (Point 2536556573, e7a9333fd1da8436)
[iohk.cardano.node.CoreId 2:Debug:31] [2019-08-21 14:06:08.98 UTC] [e7a9333] Took ledger snapshot DiskSnapshot 6 at genesis
[iohk.cardano.node.CoreId 2:Debug:31] [2019-08-21 14:06:08.98 UTC] [e7a9333] Deleted old snapshot DiskSnapshot 4
[iohk.cardano.node.CoreId 2:Debug:35] [2019-08-21 14:06:10.00 UTC] [e7a9333] Chain added block (Point 2536556585, 77b29e5407ba93bb)
[iohk.cardano.node.CoreId 2:Debug:35] [2019-08-21 14:06:10.00 UTC] [e7a9333] Block fits onto the current chain: (Point 2536556585, 77b29e5407ba93bb)
[iohk.cardano.node.CoreId 2:Debug:35] [2019-08-21 14:06:10.00 UTC] [e7a9333] Valid candidate (Point 2536556585, 77b29e5407ba93bb)
[iohk.cardano.node.CoreId 2:Debug:35] [2019-08-21 14:06:10.00 UTC] [e7a9333] Chain changed, new tip: (Point 2536556585, 77b29e5407ba93bb)
[iohk.cardano.node.CoreId 2:Debug:35] [2019-08-21 14:06:18.00 UTC] [77b29e5] Chain added block (Point 2536556589, e4eacd81a79d2eea)
[iohk.cardano.node.CoreId 2:Debug:35] [2019-08-21 14:06:18.00 UTC] [77b29e5] Block fits onto the current chain: (Point 2536556589, e4eacd81a79d2eea)
[iohk.cardano.node.CoreId 2:Debug:35] [2019-08-21 14:06:18.00 UTC] [77b29e5] Invalid block (Point 2536556589, e4eacd81a79d2eea): ExtValidationErrorOuroboros (PBftExceededSignThreshold 10 10)
[iohk.cardano.node.CoreId 2:Debug:35] [2019-08-21 14:06:18.00 UTC] [77b29e5] Invalid candidate (Point 2536556589, e4eacd81a79d2eea): InvalidBlockInSuffix
[iohk.cardano.node.CoreId 2:Debug:35] [2019-08-21 14:06:26.00 UTC] [77b29e5] Chain added block (Point 2536556593, 59a9df1be0b0ee0d)
[iohk.cardano.node.CoreId 2:Debug:35] [2019-08-21 14:06:26.00 UTC] [77b29e5] Block fits onto the current chain: (Point 2536556593, 59a9df1be0b0ee0d)
[iohk.cardano.node.CoreId 2:Debug:35] [2019-08-21 14:06:26.00 UTC] [77b29e5] Invalid block (Point 2536556593, 59a9df1be0b0ee0d): ExtValidationErrorOuroboros (PBftExceededSignThreshold 10 10)
[iohk.cardano.node.CoreId 2:Debug:35] [2019-08-21 14:06:26.00 UTC] [77b29e5] Invalid candidate (Point 2536556593, 59a9df1be0b0ee0d): InvalidBlockInSuffix
[iohk.cardano.node.CoreId 2:Debug:31] [2019-08-21 14:06:28.98 UTC] [77b29e5] Took ledger snapshot DiskSnapshot 7 at genesis
[iohk.cardano.node.CoreId 2:Debug:31] [2019-08-21 14:06:28.98 UTC] [77b29e5] Deleted old snapshot DiskSnapshot 5
[iohk.cardano.node.CoreId 2:Debug:35] [2019-08-21 14:06:34.00 UTC] [77b29e5] Chain added block (Point 2536556597, b1105a129b767095)
[iohk.cardano.node.CoreId 2:Debug:35] [2019-08-21 14:06:34.00 UTC] [77b29e5] Block fits onto the current chain: (Point 2536556597, b1105a129b767095)
[iohk.cardano.node.CoreId 2:Debug:35] [2019-08-21 14:06:34.00 UTC] [77b29e5] Invalid block (Point 2536556597, b1105a129b767095): ExtValidationErrorOuroboros (PBftExceededSignThreshold 10 10)
[iohk.cardano.node.CoreId 2:Debug:35] [2019-08-21 14:06:34.00 UTC] [77b29e5] Invalid candidate (Point 2536556597, b1105a129b767095): InvalidBlockInSuffix
[iohk.cardano.node.CoreId 2:Debug:35] [2019-08-21 14:06:42.00 UTC] [77b29e5] Chain added block (Point 2536556601, 9c55f199ae03f1f8)
[iohk.cardano.node.CoreId 2:Debug:35] [2019-08-21 14:06:42.00 UTC] [77b29e5] Block fits onto the current chain: (Point 2536556601, 9c55f199ae03f1f8)
[iohk.cardano.node.CoreId 2:Debug:35] [2019-08-21 14:06:42.00 UTC] [77b29e5] Invalid block (Point 2536556601, 9c55f199ae03f1f8): ExtValidationErrorOuroboros (PBftExceededSignThreshold 10 10)
[iohk.cardano.node.CoreId 2:Debug:35] [2019-08-21 14:06:42.00 UTC] [77b29e5] Invalid candidate (Point 2536556601, 9c55f199ae03f1f8): InvalidBlockInSuffix
```

**Comment by CodiePP** [comment]
this points to the newest revision on `master` in iohk-monitoring-framework: https://github.com/input-output-hk/iohk-monitoring-framework/pull/423

File: cardano-node/src/Cardano/Node/Tracers.hs [hunk 1]
```diff
@@ -118,8 +235,9 @@ mkTracers :: forall peer blk.
 mkTracers traceOptions tracer = Tracers
     { chainDBTracer
         = if traceChainDB traceOptions
-          then contramap show tracer'
-          else toLogObject' (traceVerbosity traceOptions) tracer
+          -- then contramap show tracer'
```
Comment (Jimbo4350) [inline cardano-node/src/Cardano/Node/Tracers.hs]: Left over comment?

Comment (CodiePP) [reply cardano-node/src/Cardano/Node/Tracers.hs]: yes! :+1:

**Pull request merged**